src/app/payment/paymentMethod/CheckoutcomCustomFormFields.tsx

```diff
@@ -68,9 +85,88 @@ const Fawry: FunctionComponent<CheckoutcomAPMFormProps> = () => {
     );
 };
 
+const Ideal: FunctionComponent<CheckoutcomAPMFormProps> = ({ method }) => {
+    const [selectedIssuer, setSelectedIssuer] = useState<Issuer | undefined>();
+    const [bicValue, setBicValue] = useState('');
+    const render = useCallback((props: FieldProps) => <HiddenInput { ...props } selectedIssuer={ bicValue } />, [bicValue]);
+
+    const issuers: Issuer[] = method.initializationData.idealIssuers;
+
+    const handleClick = ({ currentTarget }: SyntheticEvent<HTMLButtonElement>) => {
+        const issuerSelected = issuers.find(({ bic }) => bic === currentTarget?.dataset.bic);
```

Comment:
`selectedIssuer` ?

`issuerSelected` sounds more like an event name

Reply:
I also added a underscore to the variable name to avoid variable name shadowing issue with the variable above